File: Orbis_C2C_API/js/orbisapi_c2c_auth.js
import * as dotenv from "dotenv";
import { v4 as uuidv4 } from "uuid";
import jws from "jws";
import axios from "axios";

const conf = dotenv.config().parsed;

const key = jws.sign({
  header: {
    alg: "HS256",
  },
  payload: {
    group: conf.GROUP,
    entity: conf.ENTITY,
    iss: "orbis_c2c_api",
    jti: uuidv4(),
    iat: Math.floor(Date.now() / 1000),
    exp: Math.floor((Date.now() + 60 * 1000) / 1000),
  },
  secret: conf.SECRET,
});

(async () => {
  try {
    //// Get token
    const { data } = await axios.get(conf.HOSTNAME + "/c2c/jws.action?jws=" + key);
    console.log(data.token);

    //// Request
    // const res = await axios.get(conf.HOSTNAME + "/api/user/account?account=TRYK0001", {
    //   headers: {
    //     Authorization: "C2C " + Buffer.from(data.token).toString("base64"),
    //   },
    // });
    // console.log(res.data);
  } catch (err) {
    console.error(err);
  }
})();

File: Orbis_C2C_API/python/orbisapi_c2c_auth.py
import requests
import os
import json
import base64
from jose import jws
from uuid import uuid4
from datetime import datetime
from dotenv import dotenv_values

if __name__ == '__main__':
    config = dotenv_values(".env")

    payload = {
        'group': config['GROUP'],
        'entity': config['ENTITY'],
        'iss': 'C2C_API_TEST',
        'jti': str(uuid4()),
        'iat': int(datetime.now().timestamp()),
        'exp': int(datetime.now().timestamp()) + 60,
    }

    jws_token = jws.sign(payload, config['SECRET'], algorithm='HS256')

    ## Get token
    r = requests.get(f'{config["HOSTNAME"]}/api/auth/c2c/token?jws={jws_token}')
    token = json.loads(r.text)['token']
    token_bs64 = base64.b64encode(token.encode()).decode()
    print(json.loads(r.text))

    ## Request
    # r = requests.get(
    #     f'{config["HOSTNAME"]}/api/user/account?account=TRYK0001',
    #     headers={'Authorization': f'C2C {token_bs64}'},
    # )
    # print(json.loads(r.text))
